desc/optimize/_constraint_wrappers.py

```diff
@@ -98,26 +99,16 @@ def build(self, use_jit=None, verbose=1):
         self._dim_x = self._objective.dim_x
         self._dim_x_reduced = self._Z.shape[1]
 
+        # equivalent matrix for A[unfixed_idx]@Z == A@unfixed_idx_mat
+        self._unfixed_idx_mat = (
```

Comment:
So what is _unfixed_idx_mat in the above equation. Is it like a mask that picks up certain indices in A?

Reply:
`A[unfixed_idx]` are the parts of the linear constraint matrix `A` that are not just a row of all zeros with a single 1 (which are the `fixed_idx`, the indices of `x` which are just directly fixed by constraints and therefore can be just put in the particular solution and removed from the overall `x` vector, I think)

so we use `A[unfixed_idx]` a lot as it then is used to find the reduced optimization vector, I think this varibale here is just a convenience definition as we use it a lot